src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala

```diff
@@ -69,39 +65,44 @@ class BlockImporter(
       SignedTransaction.retrieveSendersInBackGround(blocks.toList.map(_.body))
       importBlocks(blocks, DefaultBlockImport)(state)
 
-    //TODO ETCM-389: Handle mined, checkpoint and new blocks uniformly
-    case MinedBlock(block) =>
-      if (!state.importing) {
-        importMinedBlock(block, state)
-      }
+    case MinedBlock(block) if !state.importing =>
+        importBlock(
+          block,
+          new MinedBlockImportMessages(block),
+          MinedBlockImport,
+          false,
+          true)(state)
 
-    //TODO ETCM-389: Handle mined, checkpoint and new blocks uniformly
-    case nc @ NewCheckpoint(parentHash, signatures) =>
+    case nc @ NewCheckpoint(block) =>
```

Comment:
wouldn't it be more consistent to have:
```scala
case nc @ NewCheckpoint(block) if state.importing =>
context.system.scheduler.scheduleOnce(1.second, self, nc)
case nc @ NewCheckpoint(block) =>
...
```
?

Reply:
totally agree, missed this one